### PR TITLE
[@mantine/dates]: Fix navigation order for date components

### PR DESCRIPTION
**Pull request by filiptammergard**
Fixes https://github.com/mantinedev/mantine/issues/3660.

DatePicker/Calendar:
- Only add selected date to tab order when month has selected date.
- Only add current date to tab order when no date is selected and current date is in month.
- Only add first non-disabled date to tab order when no date is selected and current date is not in month.

MonthPicker:
- Only add selected month to tab order when year has selected month.
- Only add current month to tab order when no month is selected and current month is in year.
- Only add first non-disabled month to tab order when no month is selected and current month is not in year.

YearPicker:
- Only add selected year to tab order when decade has selected year.
- Only add current year to tab order when no year is selected and current year is in decade.
- Only add first non-disabled year to tab order when no year is selected and current year is not in decade.

### CODE REVIEW

File: src/mantine-dates/src/components/Month/get-day-tabindex/get-day-tabindex.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const hasTodayInMonth = isSameMonth(new Date(), month);
+
+  const isToday = dayjs(date).isSame(new Date(), 'day');
+  if (!hasSelectedDayInMonth && isToday) {
+    return 0;
+  }
+
+  const isOutside = !isSameMonth(date, month);
+  const isFirstDayInMonth = dayjs(date).format('DD') === '01' && !isOutside;
+  if (!hasSelectedDayInMonth && !hasTodayInMonth && isFirstDayInMonth) {
```
Comment (rtivital) [inline src/mantine-dates/src/components/Month/get-day-tabindex/get-day-tabindex.ts:25]: What if the first day of moth is disabled? For example, it is before min date, after max date, disabled with `excludeDate` or getDayProps makes it disabled.

File: src/mantine-dates/src/components/MonthsList/get-month-tabindex/get-month-tabindex.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import dayjs from 'dayjs';
+
+export function getMonthTabIndex(month: Date, months: Date[][]) {
+  if (dayjs(month).isSame(months[0][0], 'month')) {
```
Comment (rtivital) [inline src/mantine-dates/src/components/MonthsList/get-month-tabindex/get-month-tabindex.ts:4]: - Why is there no logic for selected month?
- Same question as with days – what if first month is disabled?

**Comment by rtivital** [comment]
@filiptammergard Are you planning to finish your work on this PR?

**Comment by filiptammergard** [comment]
> @filiptammergard Are you planning to finish your work on this PR?

Yeah, I'll try to make some time for this within the next couple of days. Feel free to continue from your side if you want—the most important thing for me is that tab order is improved since I'm using the component in some projects.

File: src/mantine-dates/src/components/Month/get-day-tabindex/get-day-tabindex.ts (hunk 1 above)
Comment (filiptammergard) [reply src/mantine-dates/src/components/Month/get-day-tabindex/get-day-tabindex.ts]: With the new commit I account for these things.

File: src/mantine-dates/src/components/MonthsList/get-month-tabindex/get-month-tabindex.ts (hunk 2 above)
Comment (filiptammergard) [reply src/mantine-dates/src/components/MonthsList/get-month-tabindex/get-month-tabindex.ts]: Added the same logic for all levels now!

**Comment by filiptammergard** [comment]
Hey @rtivital! Took some time to continue on this one and changed the approach to compute tab order based on all years/months/day instead, since you cannot determine tab order based on an isolated year/month/day.

Is it missing something still?

**Comment by rtivital** [comment]
Works great, thanks for your work on this issue!

**Pull request merged**